components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java

```diff
@@ -53,6 +53,7 @@ public class ServiceProvider implements Serializable {
     private static final String IS_MANAGEMENT_APP = "IsManagementApp";
 
     private static final String IS_B2B_SELF_SERVICE_APP = "IsB2BSelfServiceApp";
+    private static final String APPLICATION_ROLE = "applicationRole";
```

Comment:
this requires changing the API contracts, have you done those where applicable? (for SOAP regenerate wsdl/service stub, for rest API resource/version etc). And is this backward compatible?

Reply:
Also, shouldn't this be an array? supporting multiple roles?

Reply:
This constant is used as the XmlElement name in https://github.com/wso2/carbon-identity-framework/pull/4871/files#diff-1ae98054c1b9ae7db11bb2177188c794ba9b91e42625ff59c59b63c9e0d48124R141. app roles is an array. We can add multiple roles

Reply:
We are not providing rest API support for this currently.

Reply:
Add service stubs for app roles